chia/wallet/derive_keys.py

```diff
@@ -110,3 +112,26 @@ def find_authentication_sk(all_sks: List[PrivateKey], owner_pk: G1Element) -> Op
                 # NOTE: ONLY use 0 for authentication key index to ensure compatibility
                 return master_sk_to_pooling_authentication_sk(sk, uint32(pool_wallet_index), uint32(0))
     return None
+
+
+def match_address_to_sk(
+    sk: PrivateKey, addresses_to_search: List[bytes32], max_ph_to_search: int = 500
+) -> Dict[bytes32, bool]:
```

Comment:
minor: As mentioned earlier, a set would be a bit more natural to work with, but not critical.